test/arranger-conduit/ReturnFunds.t.sol

```diff
@@ -1,25 +1,86 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 pragma solidity ^0.8.13;
 
-import { stdError } from "../../lib/forge-std/src/StdError.sol";
-
 import { MockERC20 } from "../../lib/mock-erc20/src/MockERC20.sol";
 
 import { IArrangerConduit } from "../../src/interfaces/IArrangerConduit.sol";
 
 import { ConduitAssetTestBase } from "./ConduitTestBase.t.sol";
+import { RevertingERC20 }       from "./RevertingERC20.sol";
 
 contract ArrangerConduit_ReturnFundsTests is ConduitAssetTestBase {
 
-    function test_returnFunds_oneRequest_exact() external {
-        asset.mint(address(this), 100);
+    function test_returnFunds_notArranger() public {
+        asset.mint(operator, 100);
+
+        vm.startPrank(operator);
         asset.approve(address(conduit), 100);
+        conduit.deposit(ilk, address(asset), 100);
+
+        vm.prank(arranger);
+        conduit.drawFunds(address(asset), 100);
+
+        vm.prank(operator);
+        conduit.requestFunds(ilk, address(asset), 100, "info");
+
+        vm.expectRevert("ArrangerConduit/not-arranger");
+        conduit.returnFunds(0, 100);
+    }
+
+    function test_returnFunds_notPending_completed() external {
+        _depositAndDrawFunds(asset, operator, ilk, 100);
+
+        vm.prank(operator);
+        conduit.requestFunds(ilk, address(asset), 100, "info");
+
+        vm.startPrank(arranger);
+
+        conduit.returnFunds(0, 100);
+
+        vm.expectRevert("ArrangerConduit/invalid-status");
+        conduit.returnFunds(0, 100);
+    }
+
+    function test_returnFunds_notPending_canceled() external {
+        _depositAndDrawFunds(asset, operator, ilk, 100);
+
+        vm.startPrank(operator);
+        conduit.requestFunds(ilk, address(asset), 100, "info");
+
+        conduit.cancelFundRequest(0);
+
+        vm.prank(arranger);
+        vm.expectRevert("ArrangerConduit/invalid-status");
+        conduit.returnFunds(0, 100);
+    }
+
+    function test_returnFunds_revertingTransfer() external {
+        _depositAndDrawFunds(asset, operator, ilk, 100);
+
+        vm.prank(operator);
+        conduit.requestFunds(ilk, address(asset), 100, "info");
 
+        RevertingERC20 erc20 = new RevertingERC20("erc20", "ERC20", 18);
+
+        vm.etch(address(asset), address(erc20).code);
```

Comment:
Here the etch makes more sense, but I think you can do this in a cleaner way by using [mocked revert call](https://book.getfoundry.sh/cheatcodes/mock-call-revert) or the [mock call](https://book.getfoundry.sh/cheatcodes/mock-call) one if you just want to return a boolean false.